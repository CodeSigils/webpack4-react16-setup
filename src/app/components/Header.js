import React from 'react';

const Header = () => (
  <header className="px4">
    <h1 className="h1">My Recipes</h1>
  </header>
);

export default Header;
