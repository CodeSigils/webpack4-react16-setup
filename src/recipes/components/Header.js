import React from 'react';
import { NavLink, Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const HeaderLink = ({ children, ...props }) => (
  <NavLink
    exact
    className="p1 mx2 black rounded text-decoration-none"
    activeClassName="bg-white"
    {...props}
  >
    {children}
  </NavLink>
);

const Header = () => (
  <header className="flex align-middle justify-between px4 m3">
    <Link className="text-decoration-none" href="/" to="/">
      <h1 className="h1 red">
        <span className="red pr1" role="img" aria-label="my recipes">
          ðŸ’—
        </span>
        My Recipes
      </h1>
    </Link>
    <nav>
      <HeaderLink to="/">Home</HeaderLink>
      <HeaderLink to="/favorites">Favorites</HeaderLink>
    </nav>
  </header>
);

HeaderLink.propTypes = { children: PropTypes.node };

export default Header;
