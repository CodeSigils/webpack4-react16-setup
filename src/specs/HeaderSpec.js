import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../recipes/components/Header';

/**
 * Think about the component in test as an input-output function.
 * Consider what props this component takes in its input, and what output
 * generates based on those props.
 * Then create a snapshot test for all possible combinations of those props.
 * In this case Header does not take any props. So we will test for proper
 * rendering.
 */
describe('<Header/>', () => {
  test('Should render correctly', () => {
    // Transform  the component into a JSON structure
    const component = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    // Tell Jest to check if tree muches the snapshot
    expect(tree).toMatchSnapshot();
  });
});
