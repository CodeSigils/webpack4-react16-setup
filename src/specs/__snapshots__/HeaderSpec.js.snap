// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Should render correctly 1`] = `
<header
  className="flex align-middle justify-between px4 m3"
>
  <a
    className="text-decoration-none"
    href="/"
    onClick={[Function]}
  >
    <h1
      className="h1 red"
    >
      <span
        aria-label="my recipes"
        className="red pr1"
        role="img"
      >
        ðŸ’—
      </span>
      My Recipes
    </h1>
  </a>
  <nav>
    <a
      aria-current={false}
      className="p1 mx2 black rounded text-decoration-none"
      href="/"
      onClick={[Function]}
      style={undefined}
    >
      Home
    </a>
    <a
      aria-current={false}
      className="p1 mx2 black rounded text-decoration-none"
      href="/favorites"
      onClick={[Function]}
      style={undefined}
    >
      Favorites
    </a>
  </nav>
</header>
`;
